### PR TITLE
[objcopy] Return an error in case of an invalid regex

### PR DESCRIPTION
**Pull request by kuilpd**
As of now, llvm-objcopy silently ignores a provided regex if it doesn't compile.

This patch adds returning an error saying that a regex couldn't be compiled, along with the compilation error message.

### CODE REVIEW

**Comment by avl-llvm** [comment]
This LGTM. Please wait a couple of days if @jh7370  has any issue.

**Review by avl-llvm — APPROVED** [review]
LGTM

**Review by jh7370 — COMMENTED** [review]
Nit: the commit/PR title's tag should be llvm-objcopy, not simply  objcopy.

Overall, the patch looks fine. Just a few small comments.

File: llvm/lib/ObjCopy/CommonConfig.cpp [hunk 1]
```diff
@@ -39,16 +39,15 @@ NameOrPattern::create(StringRef Pattern, MatchStyle MS,
                          IsPositiveMatch);
   }
   case MatchStyle::Regex: {
-    SmallVector<char, 32> Data;
-    auto AnchoredPattern =
-        ("^" + Pattern.ltrim('^').rtrim('$') + "$").toStringRef(Data);
-    auto RegEx = std::make_shared<Regex>(AnchoredPattern);
+    auto RegEx = Regex(Pattern);
```
Comment (jh7370) [inline llvm/lib/ObjCopy/CommonConfig.cpp:42]:
```suggestion
    Regex RegEx(Pattern);
```

**Review by jh7370 — APPROVED** [review]
LGTM, with nit

**Comment by MaskRay** [comment]
> Nit: the commit/PR title's tag should be llvm-objcopy, not simply objcopy.

This comment was not resolved. The subject is "[objcopy] Return an error in case of an invalid regex #74319". Don't forget it for future contributions:)

**Pull request merged**